src/bit-fields/src/bit.rs

```diff
@@ -0,0 +1,817 @@
+                let x = std::num::",stringify!($y),"::new(5).unwrap();
+                let bit = Bit::<_,0>(&x);
+                assert_eq!(bool::from(&bit),true);
+                let bit = Bit::<_,1>(&x);
+                assert_eq!(bool::from(&bit),false);
+                let bit = Bit::<_,2>(&x);
+                assert_eq!(bool::from(&bit),true);
+                ```
+            ")]
+            #[inline]
+            fn from(this: &Bit<'_, $y, P>) -> Self {
+                ((*this.0).get() & Bit::<$x, P>::MASK) != 0
+            }
+        }
+    };
+}
+bit_from!(u128, NonZeroU128);
+bit_from!(u64, NonZeroU64);
+bit_from!(u32, NonZeroU32);
+bit_from!(u16, NonZeroU16);
+bit_from!(u8, NonZeroU8);
+
+macro_rules! bit_mut_from {
+    ($x:ty,$y:ty) => {
+        // These values are only evaluated at compile-time, thus a failure can only occur at
+        // compile-time and would be immediately obvious. Thus it is safe to use arithmetic here.
+        #[allow(clippy::integer_arithmetic, clippy::as_conversions)]
+        impl<const P: u8> From<&BitMut<'_, $x, P>> for bool {
+            #[doc = concat!("
+                ```
+                use bit_fields::BitMut;
+                let mut x = 5", stringify!($x), ";
+                let bit = BitMut::<_,0>(&mut x);
+                assert_eq!(bool::from(&bit),true);
+                let bit = BitMut::<_,1>(&mut x);
+                assert_eq!(bool::from(&bit),false);
+                let bit = BitMut::<_,2>(&mut x);
+                assert_eq!(bool::from(&bit),true);
+                ```
+            ")]
+            #[inline]
+            fn from(this: &BitMut<'_, $x, P>) -> Self {
+                (*this.0 & BitMut::<$x, P>::MASK) != 0
+            }
+        }
+        // These values are only evaluated at compile-time, thus a failure can only occur at
+        // compile-time and would be immediately obvious. Thus it is safe to use arithmetic here.
+        #[allow(clippy::integer_arithmetic, clippy::as_conversions)]
+        impl<const P: u8> From<&BitMut<'_, $y, P>> for bool {
+            #[doc = concat!("
+                ```
+                use bit_fields::BitMut;
+                let mut x = std::num::",stringify!($y),"::new(5).unwrap();
+                let bit = BitMut::<_,0>(&mut x);
+                assert_eq!(bool::from(&bit),true);
+                let bit = BitMut::<_,1>(&mut x);
+                assert_eq!(bool::from(&bit),false);
+                let bit = BitMut::<_,2>(&mut x);
+                assert_eq!(bool::from(&bit),true);
+                ```
+            ")]
+            #[inline]
+            fn from(this: &BitMut<'_, $y, P>) -> Self {
+                (this.0.get() & BitMut::<$x, P>::MASK) != 0
+            }
+        }
+    };
+}
+
+bit_mut_from!(u128, NonZeroU128);
+bit_mut_from!(u64, NonZeroU64);
+bit_mut_from!(u32, NonZeroU32);
+bit_mut_from!(u16, NonZeroU16);
+bit_mut_from!(u8, NonZeroU8);
+
+// impl PartialEq<..> for Bit<..>
+macro_rules! bit_eq {
```

Comment:
`bit_eq` and `bit_mut_eq` seems the same, maybe adding type parameter to the macro is better:
```rust
macro_rules! impl_bit_eq {
   ($t:ident, $x:ty, $y:ty) => {
      impl<const P: u8> PartialEq for $t<'_, $x, P> {
         ...
```
So macros would look like:
```rust
impl_bit_eq!(Bit, u128, NonZeroU128);
impl_bit_eq!(Bit, u64, NonZeroU64);
...
impl_bit_eq!(BitMut, u128, NonZeroU128);
impl_bit_eq!(BitMut, u64, NonZeroU64);
...
```

Reply:
Due to the documentation this is quite a bit more difficult, it can be done, but might add more code to support it than it crops.

Reply:
Well, yeah, docs complicate this. Do we need docs for standard trait implementations though?
Anyway, if it requires unnecessary code to generate docs, then it is better to leave these 100ish lines of code alone.